### PR TITLE
[cli] Telemetry foundation: flush all exit paths, sanitize primitives, exit_code event

### PR DESCRIPTION
**Pull request by kajogo777**
> **Stack 1/6** — telemetry v2. All follow-ups stack on this PR.

Several `main()` paths returned raw exit codes and silently dropped the buffered telemetry batch (unknown command, `--version`, bare help, config errors). This PR makes flushing reliable and lays plumbing for the stack; **the only new event is `exit_code`, gated behind `VERCEL_CLI_TELEMETRY_V2`** (off by default until PR 6).

- Construct `TelemetryEventStore` before the global config is read; events are buffered but only *sent* once the opt-out state is known (`updateConfig()`); `save()` no-ops on empty batches
- Route every early `return N` through `finishEarly`/`finishWithExitCode`
- `--version`, bare help, and `vercel.json` checks run **before** the global config is loaded (unchanged from `main`); `finishEarly` loads the config best-effort purely for telemetry — when it is unreadable, the batch is silently dropped and the command still succeeds
- New `sanitize.ts`: `gatedToken`, `gatedFlag`, `slug`, `fp`, `ctxHash` — all later PRs emit only through these; **please focus security review here**
- `telemetryReporter` singleton (consumers land in PR 2)
- Drive-by fix for a latent crash family found while smoke-testing (and independently flagged by the review bot): the update-notification block in `main().then()` runs with `client === undefined` when the command exits before creating it (`--version`/bare help). Both branches are now guarded: `canAutoUpdate(client, …)` (auto-update) and `promptAndUpgrade(client, …)` (interactive TTY prompt — notice still prints, prompt is skipped). Reproduced on `main`.

Compatibility: `vercel --version` and bare help work with a corrupt/unreadable global config and an unwritable global dir (verified 0/18 failures under corrupt config + dead registry).

Tests: sanitizer edge cases (paths/secrets/unicode), store gating, flag-gated `exit_code`. Changeset included. No new type errors vs `main`.

### CODE REVIEW

File: packages/cli/src/util/telemetry/sanitize.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const FLAG_RE = /^--[a-z0-9-]{1,24}$/;
+const SLUG_RE = /^[a-z0-9/._-]{1,64}$/;
+
+const SLUG_PREFIXES = ['https://err.sh/', 'https://vercel.com/docs/'];
+
+/**
+ * Command-like tokens only (kebab-case, no paths or secrets).
+ * Everything else is redacted.
+ */
+export function gatedToken(value: string): string {
```
Comment (jeffsee55) [inline packages/cli/src/util/telemetry/sanitize.ts]: I don't think kebab-case only is a strong enough signal that this is not sensitive 🤔
